packages/nodes-base/test/nodes/Set/SetNode.test.ts

```diff
@@ -0,0 +1,219 @@
+								type: 'n8n-nodes-base.start',
+								typeVersion: 1,
+								position: [100, 300],
+							},
+							{
+								id: 'uuid-2',
+								parameters: {
+									values: {
+										number: [
+											{
+												name: 'value1',
+												value: 1,
+											},
+										],
+									},
+								},
+								name: 'Set',
+								type: 'n8n-nodes-base.set',
+								typeVersion: 1,
+								position: [280, 300],
+							},
+						],
+						connections: {
+							Start: {
+								main: [
+									[
+										{
+											node: 'Set',
+											type: 'main',
+											index: 0,
+										},
+									],
+								],
+							},
+						},
+					},
+				},
+				output: {
+					nodeExecutionOrder: ['Start', 'Set'],
+					nodeData: {
+						Set: [
+							[
+								{
+									value1: 1,
+								},
+							],
+						],
+					},
+				},
+			},
+			{
+				description: 'should run basic two node workflow',
+				input: {
+					// Leave the workflowData in regular JSON to be able to easily
+					// copy it from/in the UI
+					workflowData: {
+						nodes: [
+							{
+								id: 'uuid-1',
+								parameters: {},
+								name: 'Start',
+								type: 'n8n-nodes-base.start',
+								typeVersion: 1,
+								position: [100, 300],
+							},
+							{
+								id: 'uuid-2',
+								parameters: {
+									values: {
+										number: [
+											{
+												name: 'value1',
+												value: 1,
+											},
+										],
+										boolean: [
+											{
+												name: 'value2',
+												value: true,
+											},
+										],
+									},
+								},
+								name: 'Set',
+								type: 'n8n-nodes-base.set',
+								typeVersion: 1,
+								position: [280, 300],
+							},
+							{
+								id: 'uuid-3',
+								parameters: {
+									values: {
+										number: [
+											{
+												name: 'value1',
+												value: 1,
+											},
+										],
+									},
+								},
+								name: 'Set1',
+								type: 'n8n-nodes-base.set',
+								typeVersion: 1,
+								position: [280, 300],
+							},
+						],
+						connections: {
+							Start: {
+								main: [
+									[
+										{
+											node: 'Set',
+											type: 'main',
+											index: 0,
+										},
+									],
+								],
+							},
+							Set: {
+								main: [
+									[
+										{
+											node: 'Set1',
+											type: 'main',
+											index: 0,
+										},
+									],
+								],
+							},
+						},
+					},
+				},
+				output: {
+					nodeExecutionOrder: ['Start', 'Set'],
+					nodeData: {
+						Set: [
+							[
+								{
+									value1: 1,
+								},
+							],
+						],
+					},
+				},
+			},
+		];
+		const codexBase: string = 'n8n-nodes-base.';
+		const fakeLogger = {
+			log: () => {},
+			debug: () => {},
+			verbose: () => {},
+			info: () => {},
+			warn: () => {},
+			error: () => {},
+		} as ILogger;
+
+		Helpers.NodeTypes().addNode(codexBase + 'start', new Start());
```

Comment:
This could also be moved to a `setup` function, and we could use `input.workflowData.nodes` to automatically add all the nodes needed in the test.